Name: Custom Carrier String
Version: 2
Author: Choorp Studios
Description: This patch lets you set your own carrier string by using the Tweaks app.

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -11,6 +11,7 @@
 		this.airplaneModeInitialQueryDone = false;
 		this.carrierText = "Palm";
 		this.currentCarrierText = "Palm";
+		this.customCarrierString = "Palm";
 		this.airplaneModeTriggered = false;
 		this.show3GForEvdo = false;
 		this.showBlankStatusOnLimited = false;
@@ -96,7 +97,11 @@
 		
 		// Initialize and Subscribe for audio Status Notification
 		//this.initOnServerStart("com.palm.audio", this.audioInit.bind(this));
-				
+		
+		// Get custom carrier text from Tweaks
+		this.controller.serviceRequest("palm://org.webosinternals.tweaks.prefs/", {method: "get", parameters: {
+			owner: "com.choorp.custom-carrier-string", keys: ["carrierString"]}, onSuccess: this.gotCarrierString.bind(this)});
+		
 		// Initialize and Subscribe for Update Notification
 		this.initOnServerStart("com.palm.update", this.updateInit.bind(this));
 		
@@ -778,6 +783,12 @@
 	}
 },
 
+gotCarrierString: function(response) {
+	if((response) && (response.carrierString)) {
+		this.customCarrierString = response.carrierString
+	}
+},
+
 handlePowerQueryNotifications: function(payload){
 	this.handlePower.cancel();
 	Mojo.Log.error("SystemUI - handlePowerQueryNotifications %j", payload);
@@ -818,8 +829,8 @@
 				this.updateRSSIIcon("flight-mode");
 			}
 			else {
-				this.carrierText = $L("Phone offline");
-				this.currentCarrierText = "Phone offline";
+				this.carrierText = this.customCarrierString;
+				this.currentCarrierText = this.customCarrierString;
 				Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 			}							
 			Mojo.Dom.get('roaming').hide();
@@ -850,8 +861,8 @@
 				this.currentCarrierText = "Palm";
 			}
 			else {
-				this.carrierText = $L(payload.extended.networkName.escapeHTML());
-				this.currentCarrierText = payload.extended.networkName.escapeHTML();
+				this.carrierText = this.customCarrierString;
+				this.currentCarrierText = this.customCarrierString;
 			}
 			Mojo.Dom.get('carrier').innerHTML = this.carrierText;	   	
 			if(payload.extended.registration == 'home') {			
@@ -1288,8 +1299,8 @@
 				this.updateRSSIIcon("flight-mode");
 			}
 			else {
-				this.carrierText = $L("Phone offline");
-				this.currentCarrierText = "Phone offline";
+				this.carrierText = this.customCarrierString;
+				this.currentCarrierText = this.customCarrierString;
 				Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 			}
 			Mojo.Dom.get('datanet').className = 'off';
@@ -1314,8 +1325,8 @@
 					this.currentCarrierText = "Palm";
 				}
 				else {
-					this.carrierText = $L(networkMsg.networkName.escapeHTML());
-					this.currentCarrierText = networkMsg.networkName.escapeHTML();
+					this.carrierText = this.customCarrierString;
+					this.currentCarrierText = this.customCarrierString;
 				}
 				Mojo.Dom.get('carrier').innerHTML = this.carrierText;
 				this.simlocked = false;
@@ -3171,6 +3182,7 @@
 	
 	//Mojo.Log.error("UpdateClockEveryMinute " + date);
 	
+	Mojo.Dom.get('carrier').innerHTML = this.customCarrierString;
 	if(Mojo.Format.using12HrTime()) 
 		format = Mojo.Format.getFormatHash()['shortTime12'];
 	else
